packages/astro/src/server/middleware.ts

```diff
@@ -56,107 +63,124 @@ type AstroLocalsWithSentry = Record<string, unknown> & {
-          // We later on use the async iterator to read the body chunks
-          // see https://github.com/microsoft/TypeScript/issues/39051
-          const originalBody = originalResponse.body as NodeJS.ReadableStream | null;
-          if (!originalBody) {
-            return originalResponse;
-          }
-
-          const newResponseStream = new ReadableStream({
-            start: async controller => {
-              for await (const chunk of originalBody) {
-                const html = typeof chunk === 'string' ? chunk : new TextDecoder().decode(chunk);
-                const modifiedHtml = addMetaTagToHead(html, hub, span);
-                controller.enqueue(new TextEncoder().encode(modifiedHtml));
-              }
-              controller.close();
-            },
-          });
-
-          return new Response(newResponseStream, originalResponse);
+        data: {
+          method,
+          url: stripUrlQueryAndFragment(ctx.url.href),
+          ...(ctx.url.search && { 'http.query': ctx.url.search }),
+          ...(ctx.url.hash && { 'http.fragment': ctx.url.hash }),
+          ...(options.trackHeaders && { headers: allHeaders }),
         },
-      );
-      return res;
-    } catch (e) {
-      sendErrorToSentry(e);
-      throw e;
-    }
-    // TODO: flush if serveless (first extract function)
-  };
-};
+      },
+      async span => {
+        const originalResponse = await next();
+
+        if (span && originalResponse.status) {
+          span.setHttpStatus(originalResponse.status);
+        }
+
+        const hub = getCurrentHub();
+        const client = hub.getClient();
+        const contentType = originalResponse.headers.get('content-type');
+
+        const isPageloadRequest = contentType && contentType.startsWith('text/html');
+        if (!isPageloadRequest || !client) {
+          return originalResponse;
+        }
+
+        // Type case necessary b/c the body's ReadableStream type doesn't include
+        // the async iterator that is actually available in Node
+        // We later on use the async iterator to read the body chunks
+        // see https://github.com/microsoft/TypeScript/issues/39051
+        const originalBody = originalResponse.body as NodeJS.ReadableStream | null;
+        if (!originalBody) {
+          return originalResponse;
+        }
+
+        const newResponseStream = new ReadableStream({
+          start: async controller => {
+            for await (const chunk of originalBody) {
+              const html = typeof chunk === 'string' ? chunk : new TextDecoder().decode(chunk);
+              const modifiedHtml = addMetaTagToHead(html, hub, span);
+              controller.enqueue(new TextEncoder().encode(modifiedHtml));
+            }
+            controller.close();
+          },
+        });
+
+        return new Response(newResponseStream, originalResponse);
+      },
+    );
+    return res;
+  } catch (e) {
+    sendErrorToSentry(e);
+    throw e;
```

Comment:
This will change the error stack. If we want to change this, we need something like `hideStackFrames` in Node.js